### PR TITLE
Numerous minor improvements

### LINKED ISSUE #42: `build` gulp task should always run `lint` task

### PR DESCRIPTION
**Pull request by inikulin**
1)Move word-wrap to utils
2)Configure babel to not include features which require runtime for the client code (may cause issue with our current client code, @AlexanderMoskovkin can you run qunit locally, please?)
3)Rename and refactor tests for the error formatter and diffs
4)Enable erroneously disabled linting for server tests
5)Try to preserve resulting formatting in error templates
6)Always lint before build (fixes #42)

\cc @AlexanderMoskovkin @kirovboris

### CODE REVIEW

**Comment by inikulin** [comment]
@VasilyStrelyaev Please, check texts as well

File: src/reporters/errors/templates.js [hunk 1]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.timeout}</span>IFrame loading timed out.
+        `;
     },
 
     [TYPE.inIFrameTargetLoadingTimeout]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.timeout}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               'IFrame target loading timed out.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.timeout}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: oh, I seem to have missed it
Error **at** step

File: src/reporters/errors/templates.js [hunk 2]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.unhandledException}</span>Uncaught JavaScript error <code>${err.scriptErr}</code> on page.
+        `;
     },
 
     [TYPE.uncaughtJSErrorInTestCodeStep]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.unhandledException}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `Uncaught JavaScript error in test code - <code>${err.scriptErr}</code>.`;
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.unhandledException}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 3]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+        `;
     },
 
     [TYPE.storeDomNodeOrJqueryObject]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.unhandledException}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               'It is not allowed to share the DOM element, jQuery object or a function between test steps ' +
-               'via "this" object.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.unhandledException}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 4]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
 
     [TYPE.emptyFirstArgument]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `<code data-type="step-source">${escapeNewLines(err.relatedSourceCode)}</code>\n` +
-               `A target element of the <code>${err.action}</code> action has not been found in the DOM tree. ` +
-               'If this element should be created after animation or a time-consuming operation is finished, ' +
-               'use the act.waitFor action (available for use in code) to pause test execution until this element appears.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 5]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
 
     [TYPE.invisibleActionElement]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `<code data-type="step-source">${escapeNewLines(err.relatedSourceCode)}</code>\n` +
-               `A target element of the <code>${err.action}</code> action (<code>${err.element}</code>) ` +
-               'is not visible. If this element should appear when you are hovering over another element, ' +
-               'make sure that you properly recorded the Hover action.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 6]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
 
     },
 
     [TYPE.incorrectDraggingSecondArgument]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `<code data-type="step-source">${escapeNewLines(err.relatedSourceCode)}</code>\n` +
-               `"<code>${err.action || ''}</code>" action drop target is incorrect.`;
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 7]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+        `;
     },
 
     [TYPE.incorrectPressActionArgument]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `<code data-type="step-source">${escapeNewLines(err.relatedSourceCode)}</code>\n` +
-               'press action parameter contains incorrect key code.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 8]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+        `;
     },
 
     [TYPE.emptyTypeActionArgument]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `<code data-type="step-source">${escapeNewLines(err.relatedSourceCode)}</code>\n` +
-               'The type action\'s parameter text is empty.';
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.actionError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

File: src/reporters/errors/templates.js [hunk 9]
```diff
@@ -20,230 +21,305 @@ function getAssertionMsgPrefix (err) {
...
+            The <code data-type="api">type<code> action's parameter text is empty.
+        `;
     },
 
     [TYPE.unexpectedDialog]: (err) => {
-        return `${getMsgPrefix(err)}<span data-type="category">${CATEGORY.nativeDialogError}</span>Error on step ` +
-               `<span data-type="step-name">${err.stepName}</span>: ` +
-               `Unexpected system <code>${err.dialog}</code> dialog <code>${err.message}</code> appeared.`;
+        return dedent`
+            ${getMsgPrefix(err)}<span data-type="category">${CATEGORY.nativeDialogError}</span>Error on step <span data-type="step-name">${err.stepName}</span>:
```
Comment (VasilyStrelyaev) [inline src/reporters/errors/templates.js]: Error **at** step

**Comment by VasilyStrelyaev** [comment]
I won't go on - just make sure that we have "Error **at** step" everywhere

**Comment by kirovboris** [comment]
Why do we have to run "lint" before every running? It makes debug more difficult.

**Comment by inikulin** [comment]
@kirovboris This is the only way we can force linting in all testing configurations. Anyway you will have `lint` in `test` gulp-task, what will it change for you?

**Comment by inikulin** [comment]
@VasilyStrelyaev check test names, pls

**Comment by inikulin** [comment]
FPR

**Comment by kirovboris** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: test/server/diffs-test.js [hunk 10]
```diff
@@ -55,42 +55,44 @@ describe('Diffs', function () {
...
+                expected: '"str"',
+                actual:   '""',
+                marker:   ''
             };
 
             compareDiff(err, expectedDiff);
         });
 
-        it('string - string more than max output length: diff in the begin', function () {
+        it('Should truncate long value if strings differs at the beginning', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: strings differ

File: test/server/diffs-test.js [hunk 11]
```diff
@@ -107,7 +109,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('string - string more than max output length: diff in the end', function () {
+        it('Should truncate long value if strings differs at the end', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: strings differ

File: test/server/diffs-test.js [hunk 12]
```diff
@@ -213,7 +217,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('arrays - two strings - string - string more than max output length: diff in the begin', function () {
+        it('Should produce diff if array items are long strings which differs at the beginning', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: strings that differ

File: test/server/diffs-test.js [hunk 13]
```diff
@@ -234,7 +238,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('arrays - two strings - string - string more than max output length: diff in the end', function () {
+        it('Should produce diff if array items are long strings which differs at the end', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: strings that differ

File: test/server/diffs-test.js [hunk 14]
```diff
@@ -124,7 +126,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('string - string more than max output length: diff in middle', function () {
+        it('Should truncate long value if strings differs at the middle', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: strings differ in the middle

File: test/server/diffs-test.js [hunk 15]
```diff
@@ -272,7 +276,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('arrays - comparing values more than max string length', function () {
+        it('Should produce diff if array items are objects which are too long in the serialized form', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: objects that are

File: test/server/diffs-test.js [hunk 16]
```diff
@@ -340,7 +346,7 @@ describe('Diffs', function () {
             compareDiff(err, expectedDiff);
         });
 
-        it('objects - comparing values more than max string length', function () {
+        it('Should produce diff if object properties are objects which are too long in the serialized form', function () {
```
Comment (VasilyStrelyaev) [inline test/server/diffs-test.js]: objects that are

**Comment by inikulin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**